tests/bluetooth/controller/ctrl_cte_req/src/main.c

```diff
@@ -616,6 +616,751 @@ void test_cte_req_reject_inv_ll_param_peripheral_remote(void)
+		3 * PDU_DC_PAYLOAD_SIZE_MIN,
+		PDU_DC_MAX_US(3 * PDU_DC_PAYLOAD_SIZE_MIN, phy_req->rx_phys)
+	};
+	struct node_rx_pu pu = { .status = BT_HCI_ERR_SUCCESS };
+	struct pdu_data_llctrl_cte_rsp remote_cte_rsp = {};
+	struct node_tx *tx = NULL;
+	struct node_rx_pdu *ntf;
+
+	/* Prepare */
+	event_prepare(&conn);
+
+	if (!is_local && cte_req != NULL) {
+		/* Handle remote PHY update request completion and local CTE reques in the same
+		 * event.
+		 */
+
+		/* Tx Queue should have one LL Control PDU */
+		lt_rx(LL_CTE_REQ, &conn, &tx, cte_req);
+		lt_rx_q_is_empty(&conn);
+
+		/* Rx */
+		lt_tx(LL_CTE_RSP, &conn, &remote_cte_rsp);
+	} else {
+		/* Tx Queue should NOT have a LL Control PDU */
+		lt_rx_q_is_empty(&conn);
+	}
+
+	/* Done */
+	event_done(&conn);
+
+	/* There should be two host notifications, one pu and one dle */
+	ut_rx_node(NODE_PHY_UPDATE, &ntf, &pu);
+	ut_rx_pdu(LL_LENGTH_RSP, &ntf, &length_ntf);
+
+	/* Release Ntf */
+	ull_cp_release_ntf(ntf);
+
+	if (!is_local && cte_req != NULL) {
+		/* Receive notification of sampled CTE response */
+		ut_rx_pdu(LL_CTE_RSP, &ntf, &remote_cte_rsp);
+
+		/* Release Ntf */
+		ull_cp_release_ntf(ntf);
+
+		/* Release tx node */
+		ull_cp_release_tx(&conn, tx);
+	}
+
+	/* There should not be a host notifications */
+	ut_rx_q_is_empty();
+
+	CHECK_CURRENT_PHY_STATE(conn, phy_req->tx_phys, PREFER_S2_CODING, phy_req->tx_phys);
+	if (is_local) {
+		CHECK_PREF_PHY_STATE(conn, phy_req->rx_phys, phy_req->tx_phys);
+	} else {
+		CHECK_PREF_PHY_STATE(conn, PHY_1M | PHY_2M | PHY_CODED,
+				     PHY_1M | PHY_2M | PHY_CODED);
+	}
+
+	/* There is still queued CTE REQ so number of contexts is smaller by 1 than max */
+	zassert_equal(ctx_buffers_free(), ctx_num_at_end, "Free CTX buffers %d",
+		      ctx_buffers_free());
+}
+
+/**
+ * @brief The function executes PHY update procedure in central role.
+ *
+ * The main goal for the function is to run and evaluate the PHY update control procedure.
+ * In case the PHY request is remote request and there is a local CTE request then
+ * after PHY update completion CTE request is executed in the same event.
+ * In this situation the function processes verification of CTE request completion also.
+ *
+ * @param is_local        Flag informing if PHY request is local or remote.
+ * @param cte_req         Parameters of CTE request procedure. If it is NULL there were no CTE
+ *                        request.
+ * @param phy_req         Parameters of PHY update reques.
+ * @param events_at_start Number of connection events at function start.
+ * @param ctx_num_at_end  Expected number of free procedure contexts at function end.
+ */
+static void run_phy_update_central(bool is_local, struct pdu_data_llctrl_cte_req *cte_req,
+				   struct pdu_data_llctrl_phy_req *phy_req, uint8_t events_at_start,
+				   uint8_t ctx_num_at_end)
+{
+	struct pdu_data_llctrl_phy_req rsp = { .rx_phys = PHY_1M | PHY_2M | PHY_CODED,
+					       .tx_phys = PHY_1M | PHY_2M | PHY_CODED };
```

Comment:
If the PHYs here need to be the same as the PHYs in lines 675-676, consider moving to a common variable or macro.

Reply:
Changed to macro.